### PR TITLE
upload file to workspace with button

### PR DESCRIPTION
**Pull request by TemryL**
partially address https://github.com/OpenDevin/OpenDevin/issues/1379 (upload-only with button) 

starting point to add drag and drop

https://github.com/OpenDevin/OpenDevin/assets/99132608/16a040bd-6e11-4a34-b488-df3723499b8e

### CODE REVIEW

File: opendevin/server/listen.py [hunk 1]
```diff
@@ -137,6 +138,22 @@ def select_file(file: str):
     return {'code': content}
 
 
+@app.post('/api/upload-file')
+async def upload_file(file: UploadFile):
+    try:
+        workspace_base = config.get(ConfigType.WORKSPACE_BASE)
+        file_path = Path(workspace_base) / file.filename
```
Comment (rbren) [inline opendevin/server/listen.py:145]: We need to add some validation to `filename` here, to make sure it's not relative to the workspace base

**Review by rbren — CHANGES_REQUESTED** [review]
This looks rad! Just need a bit of security for filenames

**Review by xingyaoww — APPROVED** [review]
Thanks a lot! This was quick!! LGTM - tested locally and works for me.

**Comment by xingyaoww** [comment]
@TemryL oh there's some additionaly pylint issue need to be fixed: https://github.com/OpenDevin/OpenDevin/actions/runs/8860701929/job/24349032673?pr=1394

**Comment by xingyaoww** [comment]
@TemryL You check this doc: https://github.com/OpenDevin/OpenDevin/blob/main/Development.md

Once you run `make build`, i think the pre-commit hooks will be installed for you!

**Review by rbren — APPROVED** [review]

**Comment by xingyaoww** [comment]
@TemryL Just merged! Thank you for your contribution!

**Comment by TemryL** [comment]
Thanks!

**Pull request merged**